tools/Makefile.sub

```diff
@@ -1,4 +1,90 @@
-include $(JLM_ROOT)/tools/jhls/Makefile.sub
-include $(JLM_ROOT)/tools/jlc/Makefile.sub
-include $(JLM_ROOT)/tools/jlm-hls/Makefile.sub
-include $(JLM_ROOT)/tools/jlm-opt/Makefile.sub
\ No newline at end of file
+# Copyright 2022 Magnus Sjaladner <work@sjalander.com>
+# Copyright 2019 Nico Reißmann <nico.reissmann@gmail.com>
+# See COPYING for terms of redistribution.
+
+jhls_SOURCES = \
+	tools/jhls/jhls.cpp \
+
+jhls_LIBS = \
+	libtooling \
+	libhls \
+	libllvm \
+	librvsdg \
+	libutil \
+
+jhls_EXTRA_CPPFLAGS = -I$(shell $(LLVMCONFIG) --includedir)
+
+jhls_LDFLAGS = $(shell $(LLVMCONFIG) --libs core irReader) $(shell $(LLVMCONFIG) --ldflags) \
+
+$(eval $(call common_executable,jhls))
+
+jlc_SOURCES = \
+	tools/jlc/jlc.cpp \
+
+jlc_LIBS += \
+	libtooling \
+	libllvm \
+	librvsdg \
+	libutil \
+
+jlc_EXTRA_CPPFLAGS += -I$(shell $(LLVMCONFIG) --includedir)
+
+jlc_LDFLAGS += $(shell $(LLVMCONFIG) --libs core irReader) $(shell $(LLVMCONFIG) --ldflags) $(shell $(LLVMCONFIG) --system-libs)
+
+$(eval $(call common_executable,jlc))
+
+jlm-hls_SOURCES = \
```

Comment:
jhls and jlm-hls is only useful if the libhls has been compiled with -DCIRCT.
Currently, these tools print a statement if CIRCT wasn't enabled, notifying the user.
An alternative would be to not build them at all if CIRCT isn't enabled.